trino/trino.go

```diff
@@ -1943,3 +2012,19 @@ func (s *NullSlice3Map) Scan(value interface{}) error {
 	s.Valid = true
 	return nil
 }
+
+type QueryProgressCallbackResponse struct {
+	ID         string    `json:"id"`
+	QueryStats stmtStats `json:"stmtStats"`
+	Error      stmtError `json:"error"`
```

Comment:
Having an error field in it implies users might try to use it for error handling instead of the regular flow of `database/sql`. Isn't the `QueryStats.State` field enough?

Reply:
Is it weird to use it for error logging/handling? It has detailed information on what error happened, so it was useful when analyzing cause of the error.

Here is part of my code that I am using to log error.
```go
func (qc *QueryCallback) StatsUpdated(qr trino.QueryResponse) {
    ...
    if op, err := NewTrinoOperationProgress(target, &qr); err == nil {
        log := TrinoLogJson{
	      StatusCode:   op.Status,
	      ErrorCode:    "N/A",
	      ErrorMessage: "Trino Operation Progress",
	      InfoMessage:  qr.Error.Message,
	}
	err = op.Update(qc.Ctx, log)
	...
    }
}
```

And here is what gets logged in mysql table
```
{"StatusCode":"TABLE_NOT_FOUND","ErrorCode":"46","ErrorMessage":"io.trino.spi.TrinoException error - [line 3:3: Table '.......' does not exist]","InfoMessage":"[io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48) io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43) io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:1243) io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:343) io.trino.sql.tree.Table.accept(Table.java:53) io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27) io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:360) io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeFrom(StatementAnalyzer.java:2770) io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:1589) io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:343) io.trino.sql.tree.QuerySpecification.accept(QuerySpecification.java:155) io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27) io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:360) io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:370) io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:1075) io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:343) io.trino.sql.tree.Query.accept(Query.java:107) io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27) io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:360) io.trino.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:323) io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:91) io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:83) io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:263) io.trino.execution.SqlQueryExecution.\u003cinit\u003e(SqlQueryExecution.java:186) io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:768) io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:129) io.trino.$gen.Trino_353____20210715_095735_2.call(Unknown Source) com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125) com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69) com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78) java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) java.base/java.lang.Thread.run(Thread.java:832)]"}
```

Reply:
You shouldn't do error handling when checking query progress, but when reading query results. If Trino returns an error, you'll be able to read it after calling `DB.ExecContext()`, `DB.QueryContext()`. If the error happens during reading results using `Rows.Next()`, you'd get it from `Rows.Err()`, which you should always call after `Rows.Next()` returns false.

Reply:
Being able to do error handling in two different places would be confusing and error-prone.

Reply:
`error handling when checking query progress, but when reading query results` Got it thx :)

Reply:
Removed it, and added field called lastqueryState
```go
type QueryProgressCallbackResponse struct {
	ID             string
	QueryStats     stmtStats
	LastQueryState string
}